src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionInfo.cs

```diff
@@ -52,6 +52,12 @@ public ConnectionInfo(ISqlConnectionFactory factory, string ownerUri, Connection
         public DbConnection SqlConnection { get; set; }
 
         /// <summary>
+        /// The connection to the SQL database that queries will be run against.
+        /// A single instance is created and re-used after the first query is run 
+        /// </summary>
+        public DbConnection QueryConnection { get; set; }
```

Comment:
I haven't fully investigated everything, but it looks like one ConnectionInfo is created per SQL file, so this actually (surprisingly) creates one QueryConnection per SQL file. 

It looks like the same connection is reused when the same SQL file is opened in 2 separate tabs. However, when disconnecting then reconnecting, a crash occurs. I'll look into this more tomorrow.